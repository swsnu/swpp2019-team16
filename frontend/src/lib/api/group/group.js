import client from '../client';

export function createGroup({ groupId, riders, driver, from, to }) {
  return client.post(`/api/v1/group`, { groupId, riders, driver, from, to });
}

export function acceptGroup({ groupId, driverId }) {
  return client.put(`/api/v1/group/${groupId}`, driverId);
}

export function onTaxi(riderId) {
  return client.put(`/api/v1/group/taxi/riders`, riderId);
}

export function departure({ groupId, departureTime }) {
  return client.put(`/api/v1/group/taxi/departure`, { groupId, departureTime });
}

export function fetchMyGroup(groupId) {
  return client.get(`/api/v1/group/${groupId}`);
}

export function notifyDriverLocation({ driverId, groupId, locationInfo }) {
  return client.put(`/api/v1/group/taxi/location`, {
    driverId,
    groupId,
    locationInfo,
  });
}

export function confirmCost({ groupId, totalCost }) {
  return client.put(`/api/v1/group/cost`, { groupId, totalCost });
}
