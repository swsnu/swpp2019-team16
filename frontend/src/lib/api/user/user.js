import client from '../client';

export function get({ id }) {
  return client.get(`/api/v1/user/${id}`);
}

export function updatePoint({ userId, point }) {
  return client.put(`/api/v1/user/${userId}`, { point });
}

export function riderOnTaxi({ riderId }) {
  return client.put(`/api/v1/user/taxi/rider/${riderId}`);
}

export function driverGoTaxi({ driverId }) {
  return client.put(`/api/v1/user/taxi/driver/${driverId}`);
}
