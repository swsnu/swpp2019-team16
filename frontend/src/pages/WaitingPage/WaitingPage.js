import React from 'react';
import WaitingContainer from '../../containers/WaitingContainer';
import FullWidth from '../_layouts/FullWidth';

function WaitingPage() {
  return (
    <FullWidth>
      <WaitingContainer />
    </FullWidth>
  );
}
export default WaitingPage;
