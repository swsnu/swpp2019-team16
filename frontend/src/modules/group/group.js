import { createAction, handleActions } from 'redux-actions';
import { takeLatest } from 'redux-saga/effects';
import * as groupAPI from '../../lib/api/group/group';
import createRequestSaga, {
  createRequestActionTypes,
} from '../../lib/createRequestSaga';

const [
  ACCEPT_GROUP,
  ACCEPT_GROUP_SUCCESS,
  ACCEPT_GROUP_FAILURE,
] = createRequestActionTypes('group/ACCEPT_GROUP');

const [ON_TAXI, ON_TAXI_SUCCESS, ON_TAXI_FAILURE] = createRequestActionTypes(
  'group/ON_TAXI',
);

const [
  DEPARTURE,
  DEPARTURE_SUCCESS,
  DEPARTURE_FAILURE,
] = createRequestActionTypes('group/DEPARTURE');

const [
  FETCH_MY_GROUP,
  FETCH_MY_GROUP_SUCCESS,
  FETCH_MY_GROUP_FAILURE,
] = createRequestActionTypes('group/FETCH_MY_GROUP');

const [
  NOTIFY_DRIVER_LOCATION,
  NOTIFY_DRIVER_LOCATION_SUCCESS,
  NOTIFY_DRIVER_LOCATION_FAILURE,
] = createRequestActionTypes('group/NOTIFY_DRIVER_LOCATION');

const [
  CONFIRM_COST,
  CONFIRM_COST_SUCCESS,
  CONFIRM_COST_FAILURE,
] = createRequestActionTypes('group/CONFIRM_COST');

export const GROUP_CREATED = 'group/GROUP_CREATED';
export const GROUP_UPDATED = 'group/GROUP_UPDATED';
export const UNLOAD_GROUP = 'group/UNLOAD_GROUP';
export const GROUP_COST_UPDATED = 'group/GROUP_COST_UPDATED';
export const INIT_COST_CONFIRMED = 'group/INIT_COST_CONFIRMED';

export const groupCreated = createAction(
  GROUP_CREATED,
  ({ groupId, from, to }) => ({
    groupId,
    from,
    to,
  }),
);

export const groupCostUpdated = createAction(
  GROUP_COST_UPDATED,
  ({ groupId, riderCost, cost }) => ({
    groupId,
    riderCost,
    cost,
  }),
);

export const groupUpdated = createAction(
  GROUP_UPDATED,
  ({ groupId, driver, riders, fromLocation, toLocation }) => ({
    groupId,
    driver,
    riders,
    from: fromLocation,
    to: toLocation,
  }),
);

export const unloadGroup = createAction(UNLOAD_GROUP);

export const acceptGroup = createAction(
  ACCEPT_GROUP,
  ({ groupId, driverId }) => ({
    groupId,
    driverId,
  }),
);
export const onTaxi = createAction(ON_TAXI, riderId => riderId);
export const departure = createAction(
  DEPARTURE,
  ({ groupId, departureTime }) => ({
    groupId,
    departureTime,
  }),
);
export const fetchMyGroup = createAction(FETCH_MY_GROUP, groupId => groupId);
export const notifyDriverLocation = createAction(
  NOTIFY_DRIVER_LOCATION,
  ({ driverId, groupId, locationInfo }) => ({
    driverId,
    groupId,
    locationInfo,
  }),
);
export const confirmCost = createAction(
  CONFIRM_COST,
  ({ groupId, totalCost }) => ({
    groupId,
    totalCost,
  }),
);
export const initCostConfirmed = createAction(INIT_COST_CONFIRMED);

export const acceptGroupSaga = createRequestSaga(
  ACCEPT_GROUP,
  groupAPI.acceptGroup,
);
export const onTaxiSaga = createRequestSaga(ON_TAXI, groupAPI.onTaxi);
export const departureSaga = createRequestSaga(DEPARTURE, groupAPI.departure);
export const fetchMyGroupSaga = createRequestSaga(
  FETCH_MY_GROUP,
  groupAPI.fetchMyGroup,
);
export const notifyDriverLocationSaga = createRequestSaga(
  NOTIFY_DRIVER_LOCATION,
  groupAPI.notifyDriverLocation,
);
export const confirmCostSaga = createRequestSaga(
  CONFIRM_COST,
  groupAPI.confirmCost,
);

export function* groupSaga() {
  yield takeLatest(ACCEPT_GROUP, acceptGroupSaga);
  yield takeLatest(ON_TAXI, onTaxiSaga);
  yield takeLatest(DEPARTURE, departureSaga);
  yield takeLatest(FETCH_MY_GROUP, fetchMyGroupSaga);
  yield takeLatest(NOTIFY_DRIVER_LOCATION, notifyDriverLocationSaga);
  yield takeLatest(CONFIRM_COST, confirmCostSaga);
}

const initialState = {
  group: null,
  error: null,
  costConfirmed: false,
};

const group = handleActions(
  {
    [GROUP_CREATED]: (state, { payload: group }) => ({
      ...state,
      group,
    }),

    [GROUP_UPDATED]: (state, { payload: group }) => ({
      ...state,
      group,
    }),

    [UNLOAD_GROUP]: state => ({
      ...state,
      group: null,
    }),

    [GROUP_COST_UPDATED]: (state, { payload: group }) => ({
      ...state,
      group: {
        ...state.group,
        ...group,
      },
    }),

    [ACCEPT_GROUP_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),

    [ACCEPT_GROUP_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [ON_TAXI_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),

    [ON_TAXI_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [DEPARTURE_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),

    [DEPARTURE_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [FETCH_MY_GROUP_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),

    [FETCH_MY_GROUP_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [NOTIFY_DRIVER_LOCATION_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),
    [NOTIFY_DRIVER_LOCATION_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [CONFIRM_COST_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: {
        ...state.group,
        ...group,
      },
      error: null,
      costConfirmed: true,
    }),

    [CONFIRM_COST_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),
    [INIT_COST_CONFIRMED]: state => ({
      ...state,
      costConfirmed: initialState.costConfirmed,
    }),
  },
  initialState,
);

export default group;
