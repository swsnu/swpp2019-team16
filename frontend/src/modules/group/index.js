import group from './group';

export * from './group';
export default group;