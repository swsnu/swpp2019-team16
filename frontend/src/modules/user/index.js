import user from './user';

export * from './user';
export default user;
