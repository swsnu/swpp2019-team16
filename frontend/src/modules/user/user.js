import { createAction, handleActions } from 'redux-actions';
import { takeLatest, put } from 'redux-saga/effects';
import * as userAPI from '../../lib/api/user';
import createRequestSaga, {
  createRequestActionTypes,
} from '../../lib/createRequestSaga';

const TEMP_SET_USER = 'user/TEMP_SET_USER';
const SOMEONE_ON_TAXI = 'user/SOMEONE_ON_TAXI';
const GO_TAXI = 'user/GO_TAXI';
const INIT_POINT_UPDATED_FLAG = 'user/INIT_POINT_UPDATED_FLAG';
const [CHECK, CHECK_SUCCESS, CHECK_FAILURE] = createRequestActionTypes(
  'user/CHECK',
);
const [LOGOUT, LOGOUT_SUCCESS, LOGOUT_FAILURE] = createRequestActionTypes(
  'user/LOGOUT',
);
const [
  UPDATE_POINT,
  UPDATE_POINT_SUCCESS,
  UPDATE_POINT_FAILURE,
] = createRequestActionTypes('user/UPDATE_POINT');
const [
  RIDER_ON_TAXI,
  RIDER_ON_TAXI_SUCCESS,
  RIDER_ON_TAXI_FAILURE,
] = createRequestActionTypes('user/RIDER_ON_TAXI');
const [
  DRIVER_GO_TAXI,
  DRIVER_GO_TAXI_SUCCESS,
  DRIVER_GO_TAXI_FAILURE,
] = createRequestActionTypes('user/DRIVER_GO_TAXI');

export const tempSetUser = createAction(TEMP_SET_USER, user => user);
export const check = createAction(CHECK, ({ id }) => ({ id }));
export const logout = createAction(LOGOUT);
export const updatePoint = createAction(UPDATE_POINT, ({ userId, point }) => ({
  userId,
  point,
}));

export const checkSaga = createRequestSaga(CHECK, userAPI.get);
export const riderOnTaxi = createAction(RIDER_ON_TAXI, ({ riderId }) => ({
  riderId,
}));
export const goTaxi = createAction(DRIVER_GO_TAXI, ({ driverId }) => ({
  driverId,
}));
export const someoneOnTaxi = createAction(SOMEONE_ON_TAXI, ({ riderId }) => ({
  riderId,
}));
export const taxiGone = createAction(GO_TAXI);
export const initPointUpdatedFlag = createAction(INIT_POINT_UPDATED_FLAG);

const riderOnTaxiSaga = createRequestSaga(RIDER_ON_TAXI, userAPI.riderOnTaxi);
const driverGoTaxiSaga = createRequestSaga(
  DRIVER_GO_TAXI,
  userAPI.driverGoTaxi,
);

function checkFailureSaga() {
  try {
    localStorage.removeItem('user');
  } catch (e) {
    console.error('localStorage is not working');
  }
}

export function* logoutSaga() {
  try {
    localStorage.removeItem('user');
    yield put({ type: LOGOUT_SUCCESS });
  } catch (e) {
    console.error(e);
    yield put({ type: LOGOUT_FAILURE });
  }
}

export const updatePointSaga = createRequestSaga(
  UPDATE_POINT,
  userAPI.updatePoint,
);
export function* userSaga() {
  yield takeLatest(CHECK, checkSaga);
  yield takeLatest(CHECK_FAILURE, checkFailureSaga);
  yield takeLatest(LOGOUT, logoutSaga);
  yield takeLatest(UPDATE_POINT, updatePointSaga);
  yield takeLatest(RIDER_ON_TAXI, riderOnTaxiSaga);
  yield takeLatest(DRIVER_GO_TAXI, driverGoTaxiSaga);
}

const initialState = {
  user: null,
  checkError: null,
  logoutError: null,
  updatePointError: null,
  onTaxiRidersList: [],
  driverGoTaxi: false,
  onTaxiError: null,
  goTaxiError: null,
  pointUpdated: false,
};

const user = handleActions(
  {
    [TEMP_SET_USER]: (state, { payload: user }) => ({
      ...state,
      user,
    }),
    [CHECK_SUCCESS]: (state, { payload: user }) => ({
      ...state,
      user,
      checkError: null,
    }),
    [CHECK_FAILURE]: (state, { payload: error }) => ({
      ...state,
      user: null,
      checkError: error,
    }),
    [LOGOUT_SUCCESS]: state => ({
      ...state,
      user: null,
      logoutError: null,
    }),
    [LOGOUT_FAILURE]: (state, { payload: error }) => ({
      ...state,
      logoutError: error,
    }),
    [UPDATE_POINT_SUCCESS]: (state, { payload: user }) => ({
      ...state,
      user: {
        ...user,
        user,
      },
      pointUpdated: true,
      updatePointError: null,
    }),
    [UPDATE_POINT_FAILURE]: (state, { payload: error }) => ({
      ...state,
      updatePointError: error,
    }),
    [RIDER_ON_TAXI_FAILURE]: (state, { payload: error }) => ({
      ...state,
      onTaxiError: error,
    }),
    [DRIVER_GO_TAXI_FAILURE]: (state, { payload: error }) => ({
      ...state,
      goTaxiError: error,
    }),
    [SOMEONE_ON_TAXI]: (state, { payload: riderId }) => ({
      ...state,
      onTaxiRidersList: [...state.onTaxiRidersList, riderId],
    }),
    [GO_TAXI]: state => ({
      ...state,
      driverGoTaxi: true,
    }),
    [INIT_POINT_UPDATED_FLAG]: state => ({
      ...state,
      pointUpdated: initialState.pointUpdated,
    }),
  },
  initialState,
);

export default user;
