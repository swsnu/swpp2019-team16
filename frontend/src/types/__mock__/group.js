import { createMockRider, mockDriver } from './user';

export const mockGroup = {
  groupId: 1,
  driver: mockDriver,
  riders: [
    createMockRider(1, 1, 'a@gmail.com'),
    createMockRider(2, 1, 'b@gmail.com'),
    createMockRider(3, 1, 'c@gmail.com'),
  ],
  fromLocation: 'A',
  toLocation: 'B',
};
