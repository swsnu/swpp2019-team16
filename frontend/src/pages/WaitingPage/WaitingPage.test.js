import React from 'react';
import WaitingPage from './WaitingPage';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { renderWithRedux } from '../../test/utils';
import { mockRider } from '../../types/__mock__/user';
import { mockGroup } from '../../types/__mock__/group';

describe('<WaitingPage />', () => {
  const initialState = {
    user: {
      user: mockRider,
    },
    group: {
      group: mockGroup,
    },
  };

  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter>
        <Route path="/" component={WaitingPage} />
      </MemoryRouter>,
      initialState,
    );
    expect(container).toMatchSnapshot();
  });
});
