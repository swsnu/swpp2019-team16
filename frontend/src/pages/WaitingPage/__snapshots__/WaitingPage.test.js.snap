// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WaitingPage /> SHOULD match with snapshot 1`] = `
<div>
  <div>
    Waiting Page!
  </div>
</div>
`;
